modules/installation-custom-aws-vpc.adoc

```diff
@@ -0,0 +1,136 @@
+
+.Required VPC components
+
+You must provide a suitable VPC and subnets that allow communication to your
+machines.
+
+[cols="2a,7a,3a,3a",options="header"]
+|===
+
+|Component
+|AWS type
+2+|Description
+
+|VPC
+|* `AWS::EC2::VPC`
+* `AWS::EC2::VPCEndpoint`
+2+|You must provide a public VPC for the cluster to use. The VPC uses an
+endpoint that references the route tables for each subnet to improve communication with the registry that is hosted in S3.
+
+|Public subnets
+|* `AWS::EC2::Subnet`
+* `AWS::EC2::SubnetNetworkAclAssociation`
+2+|Your VPC must have public subnets for between 1 and 3 availability zones
+and associate them with appropriate Ingress rules.
+
+|Internet gateway
+|
+* `AWS::EC2::InternetGateway`
+* `AWS::EC2::VPCGatewayAttachment`
+* `AWS::EC2::RouteTable`
+* `AWS::EC2::Route`
+* `AWS::EC2::SubnetRouteTableAssociation`
+* `AWS::EC2::NatGateway`
+* `AWS::EC2::EIP`
+2+|You must have a public internet gateway, with public routes, attached to the
+VPC. In the provided templates, each public subnet has a NAT gateway with an EIP address. These NAT gateways allow cluster resources, like private-subnet instances, to reach the internet and are not required for some restricted network or proxy scenarios.
+
+.7+|Network access control
+.7+| * `AWS::EC2::NetworkAcl`
+* `AWS::EC2::NetworkAclEntry`
+2+|You must allow the VPC to access the following ports:
+h|Port
+h|Reason
+
+|`80`
+|Inbound HTTP traffic
+
+|`443`
+|Inbound HTTPS traffic
+
+|`22`
+|Inbound SSH traffic
+
+|`1024` - `65535`
+|Inbound ephemeral traffic
+
+|`0` - `65535`
+|Outbound ephemeral traffic
+
+
+|Private subnets
+|* `AWS::EC2::Subnet`
+* `AWS::EC2::RouteTable`
+* `AWS::EC2::SubnetRouteTableAssociation`
+2+|Your VPC can have a private subnets. The provided CloudFormation templates
+can create private subnets for between 1 and 3 availability zones.
+If you use private subnets, you must provide appropriate routes and tables
+for them.
+
+|===
+
+[id="installation-custom-aws-vpc-validation_{context}"]
+== VPC validation
+
+To ensure that the subnets that you provide to the installation program are suitable, it confirms the following data:
+
+* All the subnets that you specify exist.
+* You provide private subnets.
+* The subnet CIDRs belong to the machine CIDR that you specified.
+* You provide subnets for each availability zone. Each availability zone contains no more than one public and one private subnet. If you use a private cluster, provide only a private subnet for each availability zone. Otherwise, provide exactly one public and private subnet for each availability zone.
+* You provide a private subnet for each public subnet availability zone. Machines are not provisioned in availability zones that you do not provide private subnets for.
```

Comment:
`You provide a private subnet for each public subnet availability zone`

since private subnets are always required.. `You provide a public subnet for each private subnet availability zone` ??